{"ast":null,"code":"var _jsxFileName = \"/home/anaruz/www/react/crwn-cloth-learning/client/src/components/error-boundary/error-boundary.component.jsx\";\nimport React, { Component } from 'react';\nimport { jsxDEV as _jsxDEV } from \"react/jsx-dev-runtime\";\n\nclass ErrorBoundary extends Component {\n  constructor() {\n    super();\n    this.state = {\n      hasErrored: false\n    };\n  }\n\n  static getDerivedStateFromError(error) {\n    return {\n      hasErrored: true\n    };\n  }\n\n  componentDidCatch(error, info) {\n    console.log({\n      error,\n      info\n    });\n  }\n\n  render() {\n    const {\n      hasErrored\n    } = this.state;\n    const {\n      children\n    } = this.props;\n\n    if (hasErrored) {\n      return /*#__PURE__*/_jsxDEV(\"div\", {\n        children: \" Somehting went wrong, please come back again when error is fixed\"\n      }, void 0, false, {\n        fileName: _jsxFileName,\n        lineNumber: 31,\n        columnNumber: 17\n      }, this);\n    }\n\n    return children;\n  }\n\n}\n\nexport default ErrorBoundary;","map":{"version":3,"sources":["/home/anaruz/www/react/crwn-cloth-learning/client/src/components/error-boundary/error-boundary.component.jsx"],"names":["React","Component","ErrorBoundary","constructor","state","hasErrored","getDerivedStateFromError","error","componentDidCatch","info","console","log","render","children","props"],"mappings":";AAAA,OAAOA,KAAP,IAAgBC,SAAhB,QAAiC,OAAjC;;;AAIA,MAAMC,aAAN,SAA4BD,SAA5B,CAAsC;AAElCE,EAAAA,WAAW,GAAE;AAET;AAEA,SAAKC,KAAL,GAAa;AAAEC,MAAAA,UAAU,EAAG;AAAf,KAAb;AACH;;AAE8B,SAAxBC,wBAAwB,CAAEC,KAAF,EAAS;AAEpC,WAAO;AAAEF,MAAAA,UAAU,EAAG;AAAf,KAAP;AACH;;AAEDG,EAAAA,iBAAiB,CAACD,KAAD,EAAQE,IAAR,EAAa;AAE1BC,IAAAA,OAAO,CAACC,GAAR,CAAY;AAACJ,MAAAA,KAAD;AAAQE,MAAAA;AAAR,KAAZ;AACH;;AAEDG,EAAAA,MAAM,GAAE;AAER,UAAM;AAAEP,MAAAA;AAAF,QAAiB,KAAKD,KAA5B;AACA,UAAM;AAAES,MAAAA;AAAF,QAAgB,KAAKC,KAA3B;;AAEA,QAAGT,UAAH,EAAe;AAEX,0BAAQ;AAAA;AAAA;AAAA;AAAA;AAAA;AAAA,cAAR;AACH;;AAGD,WAAOQ,QAAP;AAEH;;AAhCqC;;AAqCtC,eAAeX,aAAf","sourcesContent":["import React, { Component } from 'react'\n\n\n\nclass ErrorBoundary extends Component {\n\n    constructor(){\n\n        super()\n\n        this.state = { hasErrored : false}\n    }\n\n    static getDerivedStateFromError( error) {\n\n        return { hasErrored : true}\n    }\n\n    componentDidCatch(error, info){\n\n        console.log({error, info})\n    }\n\n    render(){\n\n    const { hasErrored } = this.state\n    const { children }  = this.props\n\n    if(hasErrored) {\n        \n        return (<div> Somehting went wrong, please come back again when error is fixed</div>)\n    }\n    \n  \n    return children\n\n}\n\n}\n\n\nexport default ErrorBoundary"]},"metadata":{},"sourceType":"module"}